### PR TITLE
Bump kapp-controller to 0.33.1

### PR DESCRIPTION
**Pull request by neil-hickey**
Signed-off-by: Neil Hickey <nhickey@vmware.com>

## What this PR does / why we need it
Bump kapp-controller 0.33.1 (latest)

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Bump kapp-controller 0.33.1 (latest)
```

## Which issue(s) this PR fixes

Fixes: #

### CODE REVIEW

File: addons/packages/kapp-controller/0.33.1/bundle/.imgpkg/images.yml
Comment (blc1996) [inline addons/packages/kapp-controller/0.33.1/bundle/.imgpkg/images.yml:5]: `ghcr.io/vmware-tanzu/carvel-kapp-controller@sha256:4141b93108b1f3431e786638a115a7e0876d756e7a13acfddd7aaa3bae41e202` is the image of 0.30.1

File: addons/packages/kapp-controller/0.33.1/bundle/config/overlays/update-deployment.yaml
Comment (blc1996) [inline addons/packages/kapp-controller/0.33.1/bundle/config/overlays/update-deployment.yaml:48]: Please update this to use new image

File: addons/packages/kapp-controller/0.33.1/bundle/config/values.star
Comment (blc1996) [inline addons/packages/kapp-controller/0.33.1/bundle/config/values.star:17]: I saw some securityContext changes in upstream basefile. Please confirm that those doesn't introduce issues for accessing `/kapp-etc/pki/tls/certs/ca-bundle.crt` file.

`generateBashCmdForDNS ` was added to enable cluster DNS access for kapp-controller. Basically, it uses a init container to edit a volumeMount which has the content of `/etc`, and later attach it to kapp-controller's `/etc`

**Comment by neil-hickey** [comment]
Steps followed:

1. edit vendir.yml to change tag.
2. `make vendir-sync-package PACKAGE=kapp-controller VERSION=0.33.1`
3. `make lock-package-images PACKAGE=kapp-controller VERSION=0.33.1`

I think step 3 above needed to be `edit config/overlays/update-deployment.yaml` to update `init-kapp-controller`'s image tag BEFORE you run `make lock-package-images`

File: addons/packages/kapp-controller/0.33.1/bundle/config/values.star
Comment (neil-hickey) [reply addons/packages/kapp-controller/0.33.1/bundle/config/values.star]: hey @blc1996 What is the context that this starlark code gets run under? We removed the `kapp-controller` user group from our Dockerfile. And so I made a commit here that removes the part where you change the group, but I can't be sure if the command still works without knowing if this is run as `root` or otherwise. I don't think we realized you all were making changes to the ca-crt like this.

Comment (blc1996) [reply addons/packages/kapp-controller/0.33.1/bundle/config/values.star]: This code runs as command in initContainer (running as root). It serves the purpose of adding core DNS IP into `/etc/resolv.conf` (only root user can edit) so that kapp-controller can resolve in-cluster service URLs.

We didn't make changes to cert files, the chmod there is just to make sure that kapp-controller container have the write access to the file.

Comment (neil-hickey) [reply addons/packages/kapp-controller/0.33.1/bundle/config/values.star]: gotcha, in that case I think this should still work as is (and removing the `chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt`)

Comment (blc1996) [reply addons/packages/kapp-controller/0.33.1/bundle/config/values.star]: Basically, we want to add an entry to `/etc/resolv.conf`, but can't do it within kapp-controller container because it's not running as root.
We added a volumeMount and use a initContainer to copy everything in `/etc` to this volumeMount, and also edit `/etc/resolv.conf` to add the core DNS IP. We then attach this volumeMount to `/etc` of kapp-controller container. Kapp-controller will be read from this volumeMount. Also write to `/etc/pki/tls/certs/ca-bundle.crt`, so we need to make sure it has the right access privilage.

Comment (neil-hickey) [reply addons/packages/kapp-controller/0.33.1/bundle/config/values.star]: Actually I wonder if we need to add `chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt` back in here? @benmoss 

Let's hold off on merging until I can confirm

**Review by blc1996 — DISMISSED** [review]
LGTM

**Comment by neil-hickey** [comment]
Looks like kapp-controller has some spelling mistakes in our CRDs. I imagine someone can overwrite the blocker? This is a good prompt for us to fix them though 🥇

**Comment by neil-hickey** [comment]
hey @blc1996 we are good to go on this one!

**Comment by neil-hickey** [comment]
Hi @stmcginnis ! We have gone ahead and rebased, let us know if there's anything else needed to merge!

**Comment by stmcginnis** [comment]
Awesome, thanks @neil-hickey!! Will take a look shortly.

**Review by stmcginnis — COMMENTED** [review]
Everything looks good to me, but would love to get @seemiller to take a look.

**Review by joshrosso — APPROVED** [review]

**Pull request merged**
